var relearn_searchindex = [
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Posts",
    "content": "There is an 1-year embargo on my thesis. From 27th September, 2025, I plan to make accessible:\nmy thesis (PDF) on HAL -\u003e link on the left the thesis template on GitHub (development + issues) + Zenodo (DOI) + Overleaf (for you) -\u003e in â€œToolsâ€ the many figures (made with BioRender) -\u003e here",
    "description": "There is an 1-year embargo on my thesis. From 27th September, 2025, I plan to make accessible:\nmy thesis (PDF) on HAL -\u003e link on the left the thesis template on GitHub (development + issues) + Zenodo (DOI) + Overleaf (for you) -\u003e in â€œToolsâ€ the many figures (made with BioRender) -\u003e here",
    "tags": [],
    "title": "About my thesis",
    "uri": "/posts/thesis/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Codes",
    "content": "Allow files access to collaborators on the computing cluster Color a specific pattern in the output of find Define symbolic link Find a specific pattern in a bunsh of files Replace symbolic links by the target files Update symbolic links",
    "description": "Allow files access to collaborators on the computing cluster Color a specific pattern in the output of find Define symbolic link Find a specific pattern in a bunsh of files Replace symbolic links by the target files Update symbolic links",
    "tags": [],
    "title": "bash",
    "uri": "/codes/codes_bash/index.html"
  },
  {
    "breadcrumb": "Welcome",
    "content": "Indeed, this page is emptyâ€¦",
    "description": "Indeed, this page is emptyâ€¦",
    "tags": [],
    "title": "Posts",
    "uri": "/posts/index.html"
  },
  {
    "breadcrumb": "Welcome",
    "content": "Related to bash:\nAllow files access to collaborators on the computing cluster Color a specific pattern in the output of find Define symbolic link Find a specific pattern in a bunsh of files Replace symbolic links by the target files Update symbolic links Related to Git:\nRemove sensitive files in Git history [Tutorial] Split a GitHub repository into a main repos and a submodule Related to R:\n[Tutorial] Access two versions of R within R Studio Clean R environment butâ€¦ Increase the size of the legend keys Subset the i-th elements of a nested list Related to md / html:\nMake a table Include an HTML file issued from a R Markdown file",
    "description": "Related to bash:\nAllow files access to collaborators on the computing cluster Color a specific pattern in the output of find Define symbolic link Find a specific pattern in a bunsh of files Replace symbolic links by the target files Update symbolic links Related to Git:\nRemove sensitive files in Git history [Tutorial] Split a GitHub repository into a main repos and a submodule Related to R:\n[Tutorial] Access two versions of R within R Studio Clean R environment butâ€¦ Increase the size of the legend keys Subset the i-th elements of a nested list Related to md / html:",
    "tags": [],
    "title": "Codes",
    "uri": "/codes/index.html"
  },
  {
    "breadcrumb": "",
    "content": "Hello there !\nI am Audrey Onfroy, research engineer in computational biology. My work consists in making sequencing data express their deepest secrets ! I recently achieved my PhD thesis at Mondor Biomedical Research Institute, in CrÃ©teil, France. You can read more about my project here.\nThis website aims to be an overview of my (professional) activities:\nCodes: I used to write a lot of code snippets on sticky notes, stuck everywhere on my computer, the desk, and a bit of the wallâ€¦ Now they are accessily from everywhere and by everyone ! ğŸ“’ Tools: Here, you can find tools I have developed to simplify and speed up my daily work. I hope they can be helpful to others as well. If you have any questions, feel free to open an issue in the associated Github repository. Analyses: The computational analyses that are public, through a preprint or in a peer-reviewed journal, are there. Posts: Everything that do not fall in the previous categoriesâ€¦ Find more on LinkedIn, such as my resume, HAL, to read about my scientific contributions, and on GitHub, but everything meaningful should be in Tools or Analyses.\nLast update : 2025-06-28",
    "description": "Hello there !\nI am Audrey Onfroy, research engineer in computational biology. My work consists in making sequencing data express their deepest secrets ! I recently achieved my PhD thesis at Mondor Biomedical Research Institute, in CrÃ©teil, France. You can read more about my project here.\nThis website aims to be an overview of my (professional) activities:\nCodes: I used to write a lot of code snippets on sticky notes, stuck everywhere on my computer, the desk, and a bit of the wallâ€¦ Now they are accessily from everywhere and by everyone ! ğŸ“’ Tools: Here, you can find tools I have developed to simplify and speed up my daily work. I hope they can be helpful to others as well. If you have any questions, feel free to open an issue in the associated Github repository. Analyses: The computational analyses that are public, through a preprint or in a peer-reviewed journal, are there. Posts: Everything that do not fall in the previous categoriesâ€¦ Find more on LinkedIn, such as my resume, HAL, to read about my scientific contributions, and on GitHub, but everything meaningful should be in Tools or Analyses.",
    "tags": [],
    "title": "Welcome",
    "uri": "/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Codes",
    "content": "Remove sensitive files in Git history [Tutorial] Split a GitHub repository into a main repos and a submodule",
    "description": "Remove sensitive files in Git history [Tutorial] Split a GitHub repository into a main repos and a submodule",
    "tags": [],
    "title": "Git",
    "uri": "/codes/codes_git/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Codes",
    "content": "[Tutorial] Access two versions of R within R Studio Clean R environment butâ€¦ Increase the size of the legend keys Subset the i-th elements of a nested list",
    "description": "[Tutorial] Access two versions of R within R Studio Clean R environment butâ€¦ Increase the size of the legend keys Subset the i-th elements of a nested list",
    "tags": [],
    "title": "R",
    "uri": "/codes/codes_r/index.html"
  },
  {
    "breadcrumb": "Welcome",
    "content": "aquarius, a R package to analyse scRNA-Seq data\naudrey-onfroy.github.io/aquarius DOI: 10.5281/zenodo.15763899 HTML organizer, a tool to arrange knitted Rmd files\ngithub.com/audrey-onfroy/organize_html DOI: 10.5281/zenodo.14505634",
    "description": "aquarius, a R package to analyse scRNA-Seq data\naudrey-onfroy.github.io/aquarius DOI: 10.5281/zenodo.15763899 HTML organizer, a tool to arrange knitted Rmd files\ngithub.com/audrey-onfroy/organize_html DOI: 10.5281/zenodo.14505634",
    "tags": [],
    "title": "Tools",
    "uri": "/tools/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Codes",
    "content": "Make a table Include an HTML file issued from a R Markdown file",
    "description": "Make a table Include an HTML file issued from a R Markdown file",
    "tags": [],
    "title": "html / markdown",
    "uri": "/codes/codes_markdown/index.html"
  },
  {
    "breadcrumb": "Welcome",
    "content": "Here are listed the single-cell RNA Sequencing (scRNA-Seq) data analyses I made. The raw data and/or count matrices are available under Data. The code to make the analyses from the count matrices up to the manuscript is available under Code. The code development is tracked with GitHub but the analyses associated with the manuscript are made identifiable using Zenodo. To enable you to explore the analysis, you can access the GitHub Pages under Analysis. The index of the GitHub Pages has been built using OrganizeHTML (more information in Tools). If you have any questions, feel free to open an issue in the associated Github repository.\nReanalysis of a public dataset of hair follicle cells in physiological condition\nscRNA-Seq atlas of hair follicle cells in the context of Hidradenitis suppurativa\nData | Code | Analysis Onfroy et al. (2025), DOI: 10.1145/3736731.3746138 Data | Code | Analysis Onfroy et al. (2025), DOI: 10.1101/2025.05.28.656362 scRNA-Seq atlas of NF1-associated Peripheral Nerve Sheath Tumors (mouse)\nscRNA-Seq atlas of NF1-associated Peripheral Nerve Sheath Tumors (patients)\nData | Code | Analysis Radomska et al. (2025), DOI: In Press Data | Code | Analysis Radomska et al. (2025), DOI: In Press scRNA-Seq atlas of subcutaneous nerves in the contexte of NF1 (mouse)\n... Data | Code | Analysis Mansour, Onfroy et al. (2025), DOI: BioRXiv ...",
    "description": "Here are listed the single-cell RNA Sequencing (scRNA-Seq) data analyses I made. The raw data and/or count matrices are available under Data. The code to make the analyses from the count matrices up to the manuscript is available under Code. The code development is tracked with GitHub but the analyses associated with the manuscript are made identifiable using Zenodo. To enable you to explore the analysis, you can access the GitHub Pages under Analysis. The index of the GitHub Pages has been built using OrganizeHTML (more information in Tools). If you have any questions, feel free to open an issue in the associated Github repository.",
    "tags": [],
    "title": "Analyses",
    "uri": "/analysis/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Tags",
    "content": "",
    "description": "",
    "tags": [],
    "title": "Tag :: Bash",
    "uri": "/tags/bash/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Tags",
    "content": "",
    "description": "",
    "tags": [],
    "title": "Tag :: R",
    "uri": "/tags/r/index.html"
  },
  {
    "breadcrumb": "Welcome",
    "content": "",
    "description": "",
    "tags": [],
    "title": "Tags",
    "uri": "/tags/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  R",
    "content": "This method is suited for computer running under the Linux Ubuntu OS.\nGoal For various reasons, we want to easily switch between two (or more!) version of the R language with RStudio.\nFrom my side, I have a specific setup of R packages installed within a Singularity environment. But, I fail to access the version of R within this environment, using RStudio. So, I installed everything locally and ask RStudio to open the specific version of R.\nMethod The method consists in:\nlocate the path to R versions of interest define alias to open RStudio using each of these versions define desktop shortcuts to open RStudio in one click Locate or install the R versions of interest First, locate the folders where are installed the R versions of interest. For me, I have:\nVersion Folder 3.6.3 /usr/local/src/R-3.6.3 4.2.2 /usr/local/src/R-4.2.2 Exemple to install R version 3.6.3:\nDownload the archive associated with the version of interest there: https://cran.r-project.org/src/base/ (eg. R-3.6.3.tar.gz) version=\"3.6.3\" wget -O R-${version}.tar.gz https://cran.r-project.org/src/base/R-${version:0:1}/R-${version}.tar.gz Decompress the archive in the final folder. # Move the archive in the final folder sudo mv R-${version}.tar.gz /usr/local/src/ # Decompress the archive cd /usr/local/src/ sudo tar -zxvf R-${version}.tar.gz # Remove the archive sudo rm -f R-${version}.tar.gz # Change directory to the uncompress archive cd R-${version} Install R. sudo ./configure sudo make sudo make install Check the installed version. R --version The default R version used by RStudio is accessible by the which R command. It can be /usr/bin/R, which is a shortcut for the last version installed. If you installed first R 4.2.2 and then R 3.6.3, /usr/bin/R --version will show the 3.6.3 version.\nNote: Within RStudio, another R version can be forced by setting the RSTUDIO_WHICH_R variable in the ~/.Renviron (or ~/.profile or ~/.bashrc) file. For instance:\nRSTUDIO_WHICH_R=/usr/local/src/R-4.2.2/bin/R Define aliases In the ~/.bashrc file, create as many aliases as you have R versions of interest.\nalias r363=\"export RSTUDIO_WHICH_R='/usr/local/src/R-3.6.3/bin/R' rstudio\" alias r422=\"export RSTUDIO_WHICH_R='/usr/local/src/R-4.2.2/bin/R' rstudio\" Note: The above commands work for Ubuntu 24.04. However, to Ubuntu 20.04, a ; should be added:\nalias r363=\"export RSTUDIO_WHICH_R='/usr/local/src/R-3.6.3/bin/R' ; rstudio\" alias r422=\"export RSTUDIO_WHICH_R='/usr/local/src/R-4.2.2/bin/R' ; rstudio\" To consider the changes, source the ~/.bashrc file:\nsource ~/.bashrc You should now be able to open RStudio with R 3.6.3 (or R 4.2.2) by running the r363 (or r422) alias command in the terminal.\nNote: Using the ~/.profile file to define an alias was not working on my computer, running under Ubuntu 20.04.\nDefine shortcuts To avoid opening the terminal and typing 4 characters to open RStudio, we can define shortcuts ! The shortcut will be represented as a logo, pinned to the dashboard. For this, many tutorials exist. I followed the one from Rich Hewlett.\nCreate a .desktop file, for instance called rstudio_with_r363.desktop, containing:\n[Desktop Entry] Type=Application Terminal=false Name=RStudio R3.6.3 Icon=/path/to/a/beautiful/rstudio_pink.png Exec=bash -c 'exec bash -i \u003c\u003c\u003c \"r363\" ' Categories=Application; Details:\nTerminal=false to not open the terminal in background Name is the hover text when you pass the cursor above the icon Icon corresponds to the path to the icon Exec is a command behind the shortcut. I found it on Stackoverflow Do not forget the ; in the end You may have as many .desktop files as you have R versions of interest.\nTo add the shortcuts in the dashboard from Ubuntu, save the .desktop files in the ~/.local/share/applications folder.\nThen, pin the shortcut to the dashboard:\nfind the shortcut using its Name, in the Ubuntu Menu right click to â€˜Pin to Dashâ€™ No need to restart the computer: everything should be well recognized !\nResources:\nRich Hewlettâ€™s blog Stackoverflow",
    "description": "This method is suited for computer running under the Linux Ubuntu OS.\nGoal For various reasons, we want to easily switch between two (or more!) version of the R language with RStudio.\nFrom my side, I have a specific setup of R packages installed within a Singularity environment. But, I fail to access the version of R within this environment, using RStudio. So, I installed everything locally and ask RStudio to open the specific version of R.",
    "tags": [
      "R",
      "Bash"
    ],
    "title": "Access two versions (or more !) of R within RStudio",
    "uri": "/codes/codes_r/two_versions_of_r/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Tags",
    "content": "",
    "description": "",
    "tags": [],
    "title": "Tag :: Git",
    "uri": "/tags/git/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  Git",
    "content": "Source: https://stackoverflow.com/questions/6403601/purging-file-from-git-repo-failed-unable-to-create-new-backup\nFilter the branch to remove the sensitive files/folders (eg. FILE1 and FOLDER1) from history: git filter-branch --index-filter 'git rm -rf --cached --ignore-unmatch FILE1 FOLDER1' HEAD Note: I donâ€™t know how to adapt if there are several branches.\nPush the changes to the branch of interest (eg. main): git push --force --set-upstream origin main From StackOverflow:\nAfter filter-branch, Git keeps refs to the old commits around, in case something goes wrong.\nClean these references (eg, on the branch main): git update-ref -d refs/original/refs/heads/main",
    "description": "Source: https://stackoverflow.com/questions/6403601/purging-file-from-git-repo-failed-unable-to-create-new-backup\nFilter the branch to remove the sensitive files/folders (eg. FILE1 and FOLDER1) from history: git filter-branch --index-filter 'git rm -rf --cached --ignore-unmatch FILE1 FOLDER1' HEAD Note: I donâ€™t know how to adapt if there are several branches.\nPush the changes to the branch of interest (eg. main): git push --force --set-upstream origin main From StackOverflow:\nAfter filter-branch, Git keeps refs to the old commits around, in case something goes wrong.",
    "tags": [
      "Bash",
      "Git"
    ],
    "title": "Remove sensitive files in git history",
    "uri": "/codes/codes_git/remove_files_git/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  bash",
    "content": "Here is a code to allow the access to all files within a folder foo, in a project BAR, on the cluster, to all members having access to the BAR project:\n#!/usr/bin/bash setfacl -R -m group:BAR:rwx foo",
    "description": "Here is a code to allow the access to all files within a folder foo, in a project BAR, on the cluster, to all members having access to the BAR project:\n#!/usr/bin/bash setfacl -R -m group:BAR:rwx foo",
    "tags": [
      "Bash"
    ],
    "title": "Allow files access on cluster",
    "uri": "/codes/codes_bash/allow_access/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  R",
    "content": "Example to remove everything from the environment, except an object called sobj:\n#!/usr/bin/R rm(list = setdiff(ls(), \"sobj\"))",
    "description": "Example to remove everything from the environment, except an object called sobj:\n#!/usr/bin/R rm(list = setdiff(ls(), \"sobj\"))",
    "tags": [
      "R"
    ],
    "title": "Clean R environment but...",
    "uri": "/codes/codes_r/clean_r_env/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  bash",
    "content": "Code Example to identify all definitions of custom functions in files with the Rmd extension, and to color the names of these functions:\n#!/usr/bin/bash find . -name \"*Rmd\" -type f \\ -exec grep -E ' = function \\(' {} + | grep -vE 'FUN|%' | grep --color -P '(?\u003c=:).*(?= = fun)' Output The output contains several rows with this pattern:\nfilepath/filename.Rmd:function_name = function( â€¦.\nExplanations find . -name \"*Rmd\" -type f: in all files with the Rmd extension in the folders currently (.) accessibleâ€¦ -exec grep -E ' = function \\(' {} +: â€¦, identify all the lines containing = function( grep -vE 'FUN|%': then, remove the lines containing FUN or %. I use this to remove the function that are defined in apply function and in pipes, for instance. grep --color -P '(?\u003c=:).*(?= = func)': then, color the characters between : and = func Related (find) topics:\nFind files containing a specific pattern Apply a function to found files",
    "description": "Code Example to identify all definitions of custom functions in files with the Rmd extension, and to color the names of these functions:\n#!/usr/bin/bash find . -name \"*Rmd\" -type f \\ -exec grep -E ' = function \\(' {} + | grep -vE 'FUN|%' | grep --color -P '(?\u003c=:).*(?= = fun)' Output The output contains several rows with this pattern:\nfilepath/filename.Rmd:function_name = function( â€¦.\nExplanations find . -name \"*Rmd\" -type f: in all files with the Rmd extension in the folders currently (.) accessibleâ€¦ -exec grep -E ' = function \\(' {} +: â€¦, identify all the lines containing = function( grep -vE 'FUN|%': then, remove the lines containing FUN or %. I use this to remove the function that are defined in apply function and in pipes, for instance. grep --color -P '(?\u003c=:).*(?= = func)': then, color the characters between : and = func",
    "tags": [
      "Bash"
    ],
    "title": "Color a specific pattern in the output of `find`",
    "uri": "/codes/codes_bash/find_with_colors/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  bash",
    "content": "Pretty easy, but I never remember that providing full path is importantâ€¦\n#!/usr/bin/bash ln -s /full_path/origin/* /full_path/destination",
    "description": "Pretty easy, but I never remember that providing full path is importantâ€¦\n#!/usr/bin/bash ln -s /full_path/origin/* /full_path/destination",
    "tags": [
      "Bash"
    ],
    "title": "Define symbolic link",
    "uri": "/codes/codes_bash/define_symlinks/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  bash",
    "content": "Example to find the pattern pattern in all files with Rmd extension, in all the folders currently (.) accessible:\n#!/usr/bin/bash find . -name \"*Rmd\" -type f -exec grep -E 'pattern' {} + Related (find) topics:\nColor the output of find Apply a function to found files",
    "description": "Example to find the pattern pattern in all files with Rmd extension, in all the folders currently (.) accessible:\n#!/usr/bin/bash find . -name \"*Rmd\" -type f -exec grep -E 'pattern' {} + Related (find) topics:\nColor the output of find Apply a function to found files",
    "tags": [
      "Bash"
    ],
    "title": "Find a specific pattern in a bunsh of files",
    "uri": "/codes/codes_bash/find_pattern_files/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  R",
    "content": "We want to increase the size of the legend keys and split the legend over 3 columns, of a ggplot object called p.\nCode By expliciting the package name:\n#!/usr/bin/R p + ggplot2::guides(color = ggplot2::guide_legend(override.aes = list(size = 5), ncol = 3)) Without the package name:\n#!/usr/bin/R library(ggplot2) p + guides(color = guide_legend(override.aes = list(size = 5), ncol = 3))",
    "description": "We want to increase the size of the legend keys and split the legend over 3 columns, of a ggplot object called p.\nCode By expliciting the package name:\n#!/usr/bin/R p + ggplot2::guides(color = ggplot2::guide_legend(override.aes = list(size = 5), ncol = 3)) Without the package name:\n#!/usr/bin/R library(ggplot2) p + guides(color = guide_legend(override.aes = list(size = 5), ncol = 3))",
    "tags": [
      "R"
    ],
    "title": "Increase the size of the legend keys",
    "uri": "/codes/codes_r/ggplot_guides/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Tags",
    "content": "",
    "description": "",
    "tags": [],
    "title": "Tag :: Html",
    "uri": "/tags/html/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  R",
    "content": "Code Select all i-th elements of a nested list x, in R language:\n#!/usr/bin/R lapply(x, `[[`, i) This is useful after having used the stringr::str_split functionâ€¦\nInput Structure of the input list x:\nx â”œâ”€â”€ A â”‚Â â”œâ”€â”€ A_1 â”‚Â â”œâ”€â”€ A_2 â”‚Â â”œâ”€â”€ A_3 â”‚Â â””â”€â”€ A_4 â”œâ”€â”€ B â”‚Â â”œâ”€â”€ B_1 â”‚Â â”œâ”€â”€ B_2 â”‚Â â”œâ”€â”€ B_3 â”‚Â â””â”€â”€ B_4 â””â”€â”€ C â”œâ”€â”€ C_1 â”œâ”€â”€ C_2 â”œâ”€â”€ C_3 â””â”€â”€ C_4 Output Structure, after execution of the command (i = 2):\nx â”œâ”€â”€ A â”‚Â â””â”€â”€ A_2 â”œâ”€â”€ B â”‚Â â””â”€â”€ B_2 â””â”€â”€ C â””â”€â”€ C_2",
    "description": "Code Select all i-th elements of a nested list x, in R language:\n#!/usr/bin/R lapply(x, `[[`, i) This is useful after having used the stringr::str_split functionâ€¦\nInput Structure of the input list x:\nx â”œâ”€â”€ A â”‚Â â”œâ”€â”€ A_1 â”‚Â â”œâ”€â”€ A_2 â”‚Â â”œâ”€â”€ A_3 â”‚Â â””â”€â”€ A_4 â”œâ”€â”€ B â”‚Â â”œâ”€â”€ B_1 â”‚Â â”œâ”€â”€ B_2 â”‚Â â”œâ”€â”€ B_3 â”‚Â â””â”€â”€ B_4 â””â”€â”€ C â”œâ”€â”€ C_1 â”œâ”€â”€ C_2 â”œâ”€â”€ C_3 â””â”€â”€ C_4 Output Structure, after execution of the command (i = 2):",
    "tags": [
      "R"
    ],
    "title": "Subset the i-th elements of a nested list",
    "uri": "/codes/codes_r/select_ith_element_in_list/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Tags",
    "content": "",
    "description": "",
    "tags": [],
    "title": "Tag :: Md",
    "uri": "/tags/md/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  bash",
    "content": "Source: https://superuser.com/questions/303559/replace-symbolic-links-with-files\nCode:\n#!/usr/bin/bash set -e for link; do test -h \"$link\" || continue dir=$(dirname \"$link\") reltarget=$(readlink \"$link\") case $reltarget in /*) abstarget=$reltarget;; *) abstarget=$dir/$reltarget;; esac rm -fv \"$link\" cp -afv \"$abstarget\" \"$link\" || { # on failure, restore the symlink rm -rfv \"$link\" ln -sfv \"$reltarget\" \"$link\" } done Usage:\nfind . -type l -exec /path_to/replace_symlinks.sh {} +",
    "description": "Source: https://superuser.com/questions/303559/replace-symbolic-links-with-files\nCode:\n#!/usr/bin/bash set -e for link; do test -h \"$link\" || continue dir=$(dirname \"$link\") reltarget=$(readlink \"$link\") case $reltarget in /*) abstarget=$reltarget;; *) abstarget=$dir/$reltarget;; esac rm -fv \"$link\" cp -afv \"$abstarget\" \"$link\" || { # on failure, restore the symlink rm -rfv \"$link\" ln -sfv \"$reltarget\" \"$link\" } done Usage:\nfind . -type l -exec /path_to/replace_symlinks.sh {} +",
    "tags": [
      "Bash"
    ],
    "title": "Replace symbolic links by the target files",
    "uri": "/codes/codes_bash/replace_symlinks/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  Git",
    "content": "Context We have a local folder MAIN associated with a GitHub repository, also called MAIN. The structure is as follows:\nMAIN â”œâ”€â”€ 1_dir_A â”‚Â â”œâ”€â”€ content_AA â”‚Â â”‚Â â”œâ”€â”€ file_AAA.ext â”‚Â â”‚Â â””â”€â”€ file_AAB.ext â”‚Â â””â”€â”€ content_AB â”‚Â â”œâ”€â”€ file_ABA.ext â”‚Â â”œâ”€â”€ file_ABB.ext â”‚Â â””â”€â”€ file_ABC.ext â”œâ”€â”€ 2_dir_B â”‚Â â”œâ”€â”€ content_BA â”‚Â â”‚Â â”œâ”€â”€ file_BAA.ext â”‚Â â”‚Â â””â”€â”€ file_BAB.ext â”‚Â â””â”€â”€ content_BB â”‚Â â”œâ”€â”€ file_BBA.ext â”‚Â â””â”€â”€ file_BBB.ext â”œâ”€â”€ README.md â”œâ”€â”€ .git â””â”€â”€ .gitignore Goal We want to:\nkeep the structure of MAIN make a new folder TOTO associated with the content of 2_dir_B keep the history of 2_dir_B in the new repository (default) keep the history of 2_dir_B in the main repository define 2_dir_B as a submodule of MAIN Method Make sure that the MAIN repository is up-to-date with your local folder, using git pull and/or git push commands. Duplicate the MAIN repository Clone the MAIN repository to initate the new one: git clone https://github.com/user-name/MAIN.git TOTO Move to TOTO: cd TOTO Get all branches (optional) Get all the branches and tags from the MAIN repository, because they have not been imported using git clone: git branch -a # check the current branches git fetch origin git branch -a # check the current branches (optional) If there are SOME branches to copy, run: git checkout -b SOME origin/SOME git branch -a # check the current branches Detach from origin Detach the local repository TOTO from the MAIN repository. git remote -v # check the remote host: MAIN.git git remote rm origin git remote -v # no more host Subset the new repository Filter the directory to keep only the 2_dir_B folder. We should --force because there is no origin. git filter-repo --path 2_dir_B/ --force Move the 2_dir_B folder content to the root: git mv 2_dir_B/* . git mv 2_dir_B/.gitignore .gitignore # for the hidden files rm -Rf 2_dir_B Add the untracked files from MAIN to TOTO. This step can/should/must be done manually by using the graphical file explorerâ€¦\nCheck that TOTO contains everything present in MAIN/2_dir_B:\ndiff -r /path/to/TOTO /path/to/MAIN/2_dir_B Ensure to ignore the folders and files that should not be tracked by the git system: nano .gitignore # fill in manually Define the commit, but do not push since there is no remote yet. git status # everything should be added git add * # if necessary, for instance .gitignore git commit -m \"Filter only 2_dir_B content\" Define the new repository On GitHub, create a new empty (no README, no LICENSE) repository, for instance called TITI.\nDefine the new origin for the TOTO folder:\ngit remote add origin https://github.com/username/TITI.git git remote -v # check the remote host Push the changes. We use --force if no file has been changed but only the git history has been filtered. We use --all to include all the branches. git push --force --all origin git push --tags origin # optionally, if there are tags Define a submodule in MAIN Go to the MAIN directory: cd /path/to/MAIN Remove the 2_dir_B folder, both from tracking and locally: git rm -r 2_dir_B rm -Rf 2_dir_B/ (optional) To eventually remove this folder from the history, use git filter-branch. git filter-branch --index-filter 'git rm -rf --cached --ignore-unmatch 2_dir_B' HEAD git push origin main --force Note: I donâ€™t know how to modify this to apply it on all branches. Actually, one should use git filter-repo --path 2_dir_B/ --invert-paths to remove a folder from history. But, it removes also the remote so I have no idea how to make it works.\nDefine 2_dir_B as a submodule of MAIN. It creates a .gitmodules file. git submodule add https://github.com/username/TITI.git 2_dir_B git commit -m \"Define submodule 2_dir_B\" git push origin main # push the .gitmodules file Thatâ€™s all ! :smile:\nOutput The MAIN directory has not changed:\nMAIN â”œâ”€â”€ 1_dir_A â”‚Â â”œâ”€â”€ content_AA â”‚Â â”‚Â â”œâ”€â”€ file_AAA.ext â”‚Â â”‚Â â””â”€â”€ file_AAB.ext â”‚Â â””â”€â”€ content_AB â”‚Â â”œâ”€â”€ file_ABA.ext â”‚Â â”œâ”€â”€ file_ABB.ext â”‚Â â””â”€â”€ file_ABC.ext â”œâ”€â”€ 2_dir_B â”‚Â â”œâ”€â”€ content_BA â”‚Â â”‚Â â”œâ”€â”€ file_BAA.ext â”‚Â â”‚Â â””â”€â”€ file_BAB.ext â”‚Â â””â”€â”€ content_BB â”‚Â â”œâ”€â”€ file_BBA.ext â”‚Â â””â”€â”€ file_BBB.ext â”œâ”€â”€ README.md â”œâ”€â”€ .git â””â”€â”€ .gitignore However, on the MAIN GitHub repository, 2_dir_B points to the last commit synchronized. For instance 2_dir_B @3c8464b.\nThe TOTO folder contains:\nTOTO â”œâ”€â”€ content_BA â”‚Â â”œâ”€â”€ file_BAA.ext â”‚Â â””â”€â”€ file_BAB.ext â”œâ”€â”€ content_BB â”‚Â â”œâ”€â”€ file_BBA.ext â”‚Â â””â”€â”€ file_BBB.ext â”œâ”€â”€ .git â””â”€â”€ .gitignore On the TITI GitHub repository, there is no trace of the MAIN, but the history of all the tracked files is conserved from their creation in the MAIN repository.\nResources Useful resources:\nhttps://www.smashingmagazine.com/2014/05/moving-git-repository-new-server/ https://graphite.dev/guides/git-filter-repo",
    "description": "Context We have a local folder MAIN associated with a GitHub repository, also called MAIN. The structure is as follows:\nMAIN â”œâ”€â”€ 1_dir_A â”‚Â â”œâ”€â”€ content_AA â”‚Â â”‚Â â”œâ”€â”€ file_AAA.ext â”‚Â â”‚Â â””â”€â”€ file_AAB.ext â”‚Â â””â”€â”€ content_AB â”‚Â â”œâ”€â”€ file_ABA.ext â”‚Â â”œâ”€â”€ file_ABB.ext â”‚Â â””â”€â”€ file_ABC.ext â”œâ”€â”€ 2_dir_B â”‚Â â”œâ”€â”€ content_BA â”‚Â â”‚Â â”œâ”€â”€ file_BAA.ext â”‚Â â”‚Â â””â”€â”€ file_BAB.ext â”‚Â â””â”€â”€ content_BB â”‚Â â”œâ”€â”€ file_BBA.ext â”‚Â â””â”€â”€ file_BBB.ext â”œâ”€â”€ README.md â”œâ”€â”€ .git â””â”€â”€ .gitignore Goal We want to:",
    "tags": [
      "Git"
    ],
    "title": "Split a GitHub repository into a main repos and a submodule",
    "uri": "/codes/codes_git/split_repo_submodule_with_history/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  html / markdown",
    "content": "Code:\n\u003ctable\u003e \u003ctr\u003e \u003ctd\u003e\u003cb style=\"font-size:20px\"\u003eA\u003c/b\u003e\u003c/td\u003e \u003ctd\u003e\u003cb style=\"font-size:20px\"\u003eB\u003c/b\u003e\u003c/td\u003e \u003c/tr\u003e \u003ctr\u003e \u003ctd\u003eC\u003c/td\u003e \u003ctd\u003eD\u003c/td\u003e \u003c/tr\u003e \u003c/table\u003e Output:\nA B C D",
    "description": "Code:\n\u003ctable\u003e \u003ctr\u003e \u003ctd\u003e\u003cb style=\"font-size:20px\"\u003eA\u003c/b\u003e\u003c/td\u003e \u003ctd\u003e\u003cb style=\"font-size:20px\"\u003eB\u003c/b\u003e\u003c/td\u003e \u003c/tr\u003e \u003ctr\u003e \u003ctd\u003eC\u003c/td\u003e \u003ctd\u003eD\u003c/td\u003e \u003c/tr\u003e \u003c/table\u003e Output:\nA B C D",
    "tags": [
      "Md",
      "Html"
    ],
    "title": "Make a table",
    "uri": "/codes/codes_markdown/make_table/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  bash",
    "content": "#!/usr/bin/bash find -type l -exec bash -c \\ 'ln -f \"$(readlink -m \"$0\")\" \"$0\"' {} \\; Note: I donâ€™t know why I wrote this commandâ€¦ At some point, I may be useful !\nRelated (find) topics:\nFind files containing a specific pattern Color the output of find",
    "description": "#!/usr/bin/bash find -type l -exec bash -c \\ 'ln -f \"$(readlink -m \"$0\")\" \"$0\"' {} \\; Note: I donâ€™t know why I wrote this commandâ€¦ At some point, I may be useful !\nRelated (find) topics:\nFind files containing a specific pattern Color the output of find",
    "tags": [
      "Bash"
    ],
    "title": "Update symbolic links",
    "uri": "/codes/codes_bash/find_update_symlinks/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  CodesÂ \u003eÂ  html / markdown",
    "content": "Open in a new tab The following picture is clickable, to open the HTML file in a browser tab:\nThe code behind this picture is:\n\u003ca href=\"/pages/just_a_page.html\"\u003e \u003cimg src=\"/images/open_in_new_tab.jpg\" alt=\".\" width=\"30vw\"\u003e \u003c/a\u003e Embed the HTML page The code of the iframe below is:\n\u003ciframe width=\"100%\" height=\"2300\" src=\"/pages/just_a_page.html\" frameborder=\"0\" allow=\"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen\u003e \u003c/iframe\u003e The result is:",
    "description": "Open in a new tab The following picture is clickable, to open the HTML file in a browser tab:\nThe code behind this picture is:\n\u003ca href=\"/pages/just_a_page.html\"\u003e \u003cimg src=\"/images/open_in_new_tab.jpg\" alt=\".\" width=\"30vw\"\u003e \u003c/a\u003e Embed the HTML page The code of the iframe below is:\n\u003ciframe width=\"100%\" height=\"2300\" src=\"/pages/just_a_page.html\" frameborder=\"0\" allow=\"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen\u003e \u003c/iframe\u003e The result is:",
    "tags": [
      "Md",
      "Html"
    ],
    "title": "Include an HTML file issued from a R Markdown file",
    "uri": "/codes/codes_markdown/include_html/index.html"
  },
  {
    "breadcrumb": "WelcomeÂ \u003eÂ  Posts",
    "content": "Iâ€™ve been planning to build my GitHub Pages for a while - and I finally did it !\nThe current theme is: hugo-theme-relearn.",
    "description": "Iâ€™ve been planning to build my GitHub Pages for a while - and I finally did it !\nThe current theme is: hugo-theme-relearn.",
    "tags": [],
    "title": "Welcome to my GitHub Pages !",
    "uri": "/posts/welcome/index.html"
  }
]
